hudi-sync/hudi-sync-common/src/main/java/org/apache/hudi/sync/common/AbstractSyncTool.java

```diff
@@ -17,17 +17,31 @@
 
 package org.apache.hudi.sync.common;
 
+import org.apache.hudi.common.config.TypedProperties;
+
+import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.fs.FileSystem;
 
 import java.util.Properties;
 
+/**
+ * Base class to sync Hudi meta data with Metastores to make
+ * Hudi table queryable through external systems.
+ */
 public abstract class AbstractSyncTool {
-  protected Properties props;
-  protected FileSystem fileSystem;
+  protected final Configuration conf;
+  protected final FileSystem fs;
+  protected TypedProperties props;
 
-  public AbstractSyncTool(Properties props, FileSystem fileSystem) {
+  public AbstractSyncTool(TypedProperties props, Configuration conf, FileSystem fs) {
```

Comment:
I think a cleaner interface would be 

```suggestion
  public AbstractSyncTool(HoodieSyncConfig syncConfig, FileSystem fs) {
```

`hadoopConf` should be set by `fs.getConf()`. And `props` is not controllable; hard for user to decide what to pass in here. 

`HoodieSyncConfig` is the matching config class for `SyncTool` as the names suggest. Internally we can still do `props = syncConfig.getProps();`